src/endian.h

```diff
@@ -0,0 +1,28 @@
+#ifndef UDX_ENDIAN_H
+#define UDX_ENDIAN_H
+
+#include "../include/udx.h"
```

Comment:
I can't remember what the best practice is, but shouldn't this also include stdint as uint32 is explictly used below rather than rely on udx.h to do so?

Reply:
stdbool also

Reply:
That practice seems to have quite a following, yes, as also evidenced by this tool: https://github.com/include-what-you-use/include-what-you-use

I'll adjust.